dingo/preprocess.py

```diff
@@ -0,0 +1,245 @@
+        self._zero_flux_reactions = self._zero_flux()
+        self._blocked_reactions = self._blocked()
+        self._mle_reactions = self._metabolically_less_efficient()
+        self._removed_reactions = []
+
+
+    def _objective_function(self):
+        """
+        A function used to find the objective function of a model
+        """
+        
+        objective = str(self._model.summary()._objective)
+        objective = objective.split(" ")[1]
+        
+        self._objective = objective
+        return self._objective
+
+
+    def _initial(self):
+        """
+        A function used to find reaction ids of a model
+        """
+        
+        self._initial_reactions = []
+        
+        for reaction in self._model.reactions:
+            reaction_id = reaction.id
+            self._initial_reactions.append(reaction_id)
+            
+        return self._initial_reactions
+    
+    
+    def _reaction_bounds_dictionary(self):
+        """
+        A function used to create a dictionary that maps
+        reactions with their corresponding bounds. It is used to
+        later restore bounds to their wild-type values
+        """
+        
+        self._reaction_bounds_dict = {
+                  'reaction': (0, 1000)
+                  }
+                
+        for reaction_id in self._initial_reactions:
+            bounds = self._model.reactions.get_by_id(reaction_id).bounds
+            self._reaction_bounds_dict[reaction_id] = bounds
+            
+        return self._reaction_bounds_dict
+
+
+    def _essentials(self):
+        """
+        A function used to find all the essential reactions
+        and append them into a list
+        """
+        
+        self._essential_reactions = []
+        essential_reactions = cobra.flux_analysis.find_essential_reactions(self._model)
+        for reaction in essential_reactions:
+            reaction_id = reaction.id
+            self._essential_reactions.append(reaction_id)
+            
+        return self._essential_reactions
+    
+
+    def _zero_flux(self):
+        """
+        A function used to find zero-flux reactions.
+        These reactions are the ones that have a flux equaled to 0
+        when running a FVA analysis with the fraction of optimum set to 90%
+        """
+        
+        tol = 1e-6
+        
+        fva = cobra.flux_analysis.flux_variability_analysis(self._model, fraction_of_optimum=0.9)
+        zero_flux = fva.loc[ (abs(fva['minimum']) < tol ) & (abs(fva['maximum']) < tol)]
+        zero_flux_reactions = zero_flux.index.tolist()
+        
+        self._zero_flux_reactions = zero_flux_reactions
+        return self._zero_flux_reactions
+    
+    
+    def _blocked(self):
+        """
+        A function used to find blocked reactions.
+        These reactions can not have any flux other than 0
+        """
+        
+        blocked_reactions = cobra.flux_analysis.find_blocked_reactions(self._model)
```

Comment:
I suggest you allow the user to set the `open_exchanges` argument of the `find_blocked_reactions()`. 

This argument allows to open all exchange reactions, i.e. all exchange reactions available in the model will be set with non-zero boundaries making any nutrient available for the model.